File: 2. Loop Problems/Loop problems in python/p19.py
# // Write a python program to find all prime factors of a number.

import math


n = int(input("Enter a number: "))


while n % 2 == 0:
    print("2", end=" ")
    n //= 2


for i in range(3, int(math.sqrt(n)) + 1, 2):
    while n % i == 0:
        print(i, end=" ")
        n //= i


if n > 2:
    print(n, end=" ")

File: 2. Loop Problems/Loop Problems in c/p19.c
// Write a C program to find all prime factors of a number.



#include <stdio.h>
#include <math.h>

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);

    
    while (n % 2 == 0) {
        printf("2 ");
        n /= 2;
    }
    for (int i =  3 ; i <= sqrt(n);i+=2) {
        while (n % i == 0) {
            printf("%d ",i);
            n /= i;
        }
    
    }
if ( n > 2) {
        printf("%d ",n);
    }
}
    
